### PR TITLE
[EventGrid] Bump the version of Microsoft.Azure.Management.EventGrid SDK client from 3.0.0-preview to 3.0.1-preview

### PR DESCRIPTION
**Pull request by ahamad-MS**
## Description

Bump the version of Microsoft.Azure.Management.EventGrid SDK client from 3.0.0-preview to 3.0.1-preview in order to account for new Newtonsoft Json dependency of 10.0.3
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/EventGrid/management/Management.EventGrid/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure EventGrid Management Library")]
 [assembly: AssemblyDescription("Provides developers with a library to create and manage all Azure EventGrid resources.")]
 
-[assembly: AssemblyVersion("3.0.0.0")]
-[assembly: AssemblyFileVersion("3.0.0.0")]
+[assembly: AssemblyVersion("3.0.1.0")]
```
Comment (dsgouda) [inline src/SDKs/EventGrid/management/Management.EventGrid/Properties/AssemblyInfo.cs]: Please do not modify `AssemblyVersion` here

File: src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj [hunk 2]
```diff
@@ -6,7 +6,7 @@
   <PropertyGroup>
     <PackageId>Microsoft.Azure.Management.EventGrid</PackageId>
     <Description>Provides developers with a library to create and manage all Azure EventGrid resources. </Description>
-    <Version>3.0.0-preview</Version>
+    <Version>3.0.1-preview</Version>
     <AssemblyName>Microsoft.Azure.Management.EventGrid</AssemblyName>
     <PackageTags>Microsoft Azure EventGrid Management;Event Grid;Event Grid management;</PackageTags>
```
Comment (dsgouda) [inline src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj]: Please update `PackageReleaseNotes` here

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please delete the `.nupkg` files

File: src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj (hunk 2 above)
Comment (ahamad-MS) [reply src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj]: Fixed

File: src/SDKs/EventGrid/management/Management.EventGrid/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (ahamad-MS) [reply src/SDKs/EventGrid/management/Management.EventGrid/Properties/AssemblyInfo.cs]: Fixed

**Review by dsgouda — APPROVED** [review]
LGTM
Please squash commits into a single commit and we are good to merge

File: src/SDKs/EventGrid/management/Management.EventGrid/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (dsgouda) [reply src/SDKs/EventGrid/management/Management.EventGrid/Properties/AssemblyInfo.cs]: `AssemblyFileVersion` must be bumped

**Pull request closed without merging**